From 5d58d469fa055575c445f2d9a2f644e915edb7fd Mon Sep 17 00:00:00 2001
From: Mingcong Bai <jeffbai@aosc.io>
Date: Tue, 10 Dec 2024 17:55:36 +0800
Subject: [PATCH 241/301] AOSCOS: drm: radeon: declare prototype for
 radeon_recover_callback()

Declare radeon_recover_callback() in radeon_device.h to clean up function
reference and avoid build errors with -Werror=missing-prototype.

Fixes: "AOSCOS: drm/radeon: recover the GPU if it fails at resume"
Signed-off-by: Mingcong Bai <jeffbai@aosc.io>

Signed-off-by: Kexy Biscuit <kexybiscuit@aosc.io>
---
 drivers/gpu/drm/radeon/radeon_device.h  | 2 ++
 drivers/gpu/drm/radeon/radeon_irq_kms.c | 3 +--
 2 files changed, 3 insertions(+), 2 deletions(-)

diff --git a/drivers/gpu/drm/radeon/radeon_device.h b/drivers/gpu/drm/radeon/radeon_device.h
index 31a0ae295cc8..e7f6d468b567 100644
--- a/drivers/gpu/drm/radeon/radeon_device.h
+++ b/drivers/gpu/drm/radeon/radeon_device.h
@@ -29,4 +29,6 @@
 
 bool radeon_device_is_virtual(void);
 
+void radeon_recover_callback(struct work_struct *work);
+
 #endif                         /* __RADEON_DEVICE_H__ */
diff --git a/drivers/gpu/drm/radeon/radeon_irq_kms.c b/drivers/gpu/drm/radeon/radeon_irq_kms.c
index e5d5ea218579..d4ad1fa82645 100644
--- a/drivers/gpu/drm/radeon/radeon_irq_kms.c
+++ b/drivers/gpu/drm/radeon/radeon_irq_kms.c
@@ -40,6 +40,7 @@
 #include "radeon_kms.h"
 #include "radeon_reg.h"
 
+#include "radeon_device.h"
 
 #define RADEON_WAIT_IDLE_TIMEOUT 200
 
@@ -307,8 +308,6 @@ static bool radeon_msi_ok(struct radeon_device *rdev)
 	return true;
 }
 
-extern void radeon_recover_callback(struct work_struct *work);
-
 /**
  * radeon_irq_kms_init - init driver interrupt info
  *
-- 
2.48.1

